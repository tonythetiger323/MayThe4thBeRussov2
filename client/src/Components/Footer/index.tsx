import * as React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import  Link  from '@material-ui/core/Link'
import  Button from '@material-ui/core/Button'
import NavbarLink from '../NavbarLink'

const styles = (theme: any) => ({
  root: {
    fontFamily: 'Oxygen',
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  }
})

interface FooterProps {
  classes: any
}

const Footer: React.FunctionComponent<FooterProps> = (props: FooterProps) => {
  const { classes } = props

  return (
    <div className={classes.root}>
      <Grid container justify='center'>
        <Grid item>
          <span style={{fontSize: 'small'}}>
              &copy; {new Date().getFullYear()} Copyright:{' '}
              <Button style={{fontSize: 'x-small'}} href='https://www.tonygreeley.com' component={Link as any}>Tony Greeley</Button>
              <NavbarLink style={{fontSize: 'x-small'}} to='/login'>Login</NavbarLink>
              <NavbarLink style={{fontSize: 'x-small'}} to='/rsvpdashboard'>Dashboard</NavbarLink>
          </span>
        </Grid>
      </Grid>
    </div>
  )
}
export default withStyles(styles)(Footer)
