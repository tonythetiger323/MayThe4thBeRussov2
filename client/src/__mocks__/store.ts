const myMock = jest.fn()
const store = new myMock()
export default store
