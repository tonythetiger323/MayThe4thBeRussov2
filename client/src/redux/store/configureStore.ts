import { createStore, compose, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history'
import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'connected-react-router'
import createRootReducer from '../reducers';
import rootSaga from '../sagas';

export const history = createBrowserHistory();

export default function configureStore(preloadedState?: any) {
  const composeEnhancer: typeof compose = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
  const sagaMiddleware = createSagaMiddleware();
  const middleware = [routerMiddleware(history), sagaMiddleware]
  const store = createStore(
    createRootReducer(history),
    preloadedState,
    composeEnhancer(
      applyMiddleware(...middleware)
    )
  )
  
  sagaMiddleware.run(rootSaga)

  return store
}
