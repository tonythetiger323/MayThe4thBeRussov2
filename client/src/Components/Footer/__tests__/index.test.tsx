import * as React from 'react'
import { shallow } from 'enzyme'
import Footer from '../../Footer'
import { createShallow } from '@material-ui/core/test-utils'
import Paper from '@material-ui/core/Paper'
import store from '../../../redux/store/index'

let wrapper: any
let shallowMui: any
describe('Footer Component', () => {
  const props = { store, isAuthenticated: false, logoutUser: '', user: 'test' }
  beforeEach(() => (wrapper = shallow(<Footer {...props} />)))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('<Paper/>', () => {
  beforeAll(() => {
    shallowMui = createShallow()
  })

  it('should work', () => {
    wrapper = shallowMui(<Paper />)
  })
})
