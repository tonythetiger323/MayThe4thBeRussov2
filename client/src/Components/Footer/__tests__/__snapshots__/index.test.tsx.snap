// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render correctly 1`] = `
<Footer
  classes={
    Object {
      "root": "Footer-root-1",
    }
  }
/>
`;
